sycl-fusion/jit-compiler/lib/translation/KernelTranslation.cpp

```diff
@@ -24,21 +24,19 @@ using namespace jit_compiler::translation;
 using namespace llvm;
 
 ///
-/// Get an attribute value consisting of NumValues scalar constant integers
-/// from the MDNode.
-static void getAttributeValues(std::vector<std::string> &Values, MDNode *MD) {
-  for (const auto &MDOp : MD->operands()) {
-    auto *ConstantMD = cast<ConstantAsMetadata>(MDOp);
-    auto *ConstInt = cast<ConstantInt>(ConstantMD->getValue());
-    Values.push_back(std::to_string(ConstInt->getZExtValue()));
-  }
+/// Get an `Indices` object from the MDNode's three constant integer operands.
+static Indices getAttributeValues(MDNode *MD) {
+  assert(MD->getNumOperands() == Indices::size());
+  Indices Res;
+  std::transform(MD->op_begin(), MD->op_end(), Res.begin(),
+                 [](const auto &MDOp) {
+                   auto *ConstantMD = cast<ConstantAsMetadata>(MDOp);
+                   auto *ConstInt = cast<ConstantInt>(ConstantMD->getValue());
+                   return ConstInt->getZExtValue();
+                 });
```

Comment:
Don't we have a function for this?

Reply:
(`getUint<size_t>`)?

Reply:
Not yet, I'll extract a helper into `MDParsing.h` after this PR series.